From 9cdb8520049629271ad411ac91ab1bea3e1cfa2b Mon Sep 17 00:00:00 2001
From: Monam Agarwal <monamagarwal123@gmail.com>
Date: Sun, 23 Mar 2014 23:58:27 +0530
Subject: [PATCH 295/454] dm: use RCU_INIT_POINTER instead of
 rcu_assign_pointer in __unbind
Git-commit: 9cdb8520049629271ad411ac91ab1bea3e1cfa2b
Patch-mainline: v3.15-rc1
References: bsc#1020795

Replace rcu_assign_pointer(p, NULL) with RCU_INIT_POINTER(p, NULL).

The rcu_assign_pointer() ensures that the initialization of a structure
is carried out before storing a pointer to that structure.  And in the
case of the NULL pointer, there is no structure to initialize.  So,
rcu_assign_pointer(p, NULL) can be safely converted to
RCU_INIT_POINTER(p, NULL).

Signed-off-by: Monam Agarwal <monamagarwal123@gmail.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/md/dm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index 0d52f6f..6382213 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -2163,7 +2163,7 @@ static struct dm_table *__unbind(struct mapped_device *md)
 		return NULL;
 
 	dm_table_event_callback(map, NULL, NULL);
-	rcu_assign_pointer(md->map, NULL);
+	RCU_INIT_POINTER(md->map, NULL);
 	dm_sync_table(md);
 
 	return map;
-- 
2.10.2

